crml/cennzx-spot/src/types.rs

```diff
@@ -44,6 +58,39 @@ impl FeeRate {
 		N::sa(lhs.as_() * SCALE_FACTOR / rhs.0)
 	}
 
+	/// Divide a `u128` supported numeric by a FeeRate
+	pub fn safe_div<N: Into<u128>>(lhs: N, rhs: FeeRate) -> rstd::result::Result<u128, &'static str> {
+		let lhs = lhs.into();
+		let lhs = U256::from(lhs);
+		let scale_factor = U256::from(SCALE_FACTOR);
+		let rhs = U256::from(rhs.0);
+		let res: Result<u128, &'static str> = (lhs * scale_factor / rhs).try_into();
```

Comment:
`let res: u128 = (lhs * scale_factor / rhs).try_into().ok_or("Overflow error")?;`

Reply:
made the error handling in this format everywhere in the code